src/selectors/tests/getCallStackFrames.spec.js

```diff
@@ -0,0 +1,228 @@
+        ],
+        sources: fromJS({
+          source1: {
+            id: "source1",
+            url: "webpack:///~/react-dom/lib/ReactCompositeComponent.js"
+          },
+          source2: { id: "source2", url: "webpack:///src/App.js" }
+        }),
+        selectedSource: fromJS({
+          id: "sourceId-originalSource"
+        })
+      };
+
+      const frames = getCallStackFrames.resultFunc(
+        state.frames,
+        state.sources,
+        state.selectedSource,
+        false
+      );
+
+      expect(frames).toEqual([
+        expect.objectContaining({ id: "frame1" }),
+        expect.objectContaining({ id: "frame2" }),
+        expect.objectContaining({ id: "frame3" })
+      ]);
+    });
+
+    it("groups library frames", () => {
+      const state = {
+        frames: [
+          { location: { sourceId: "source1" } },
+          { location: { sourceId: "source2" } },
+          { location: { sourceId: "source2" } }
+        ],
+        sources: fromJS({
+          source1: { id: "source2", url: "webpack:///src/App.js" },
+          source2: {
+            id: "source1",
+            url: "webpack:///~/react-dom/lib/ReactCompositeComponent.js"
+          }
+        }),
+        selectedSource: fromJS({
+          id: "sourceId-originalSource"
+        })
+      };
+
+      const frames = getCallStackFrames.resultFunc(
+        state.frames,
+        state.sources,
+        state.selectedSource,
+        true
+      );
+
+      expect(frames).toEqual([
+        expect.objectContaining(state.frames[0]),
+        [
+          expect.objectContaining({ ...state.frames[1], library: "React" }),
+          expect.objectContaining({ ...state.frames[2], library: "React" })
+        ]
+      ]);
+    });
+
+    it("groups the webpackBootstrap frame with other Webpack frames", () => {
+      const state = {
+        frames: [
+          { location: { sourceId: "source1" } },
+          { location: { sourceId: "source1" } },
+          // webpackBootstrap frame
+          { location: { sourceId: "source2" } }
+        ],
+        sources: fromJS({
+          source1: {
+            id: "source1",
+            url: "webpack:///webpack/bootstrap 01d88449ca6e9335a66f"
+          },
+          source2: { id: "source2", url: "https://foo.com/bundle.js" }
+        }),
+        selectedSource: fromJS({
+          id: "sourceId-originalSource"
+        })
+      };
+
+      const frames = getCallStackFrames.resultFunc(
+        state.frames,
+        state.sources,
+        state.selectedSource,
+        true
+      );
+
+      expect(frames).toEqual([
+        [
+          expect.objectContaining(state.frames[0]),
+          expect.objectContaining(state.frames[1]),
+          expect.objectContaining(state.frames[2])
+        ]
+      ]);
+    });
+
+    describe("groups Babel frames for transforming async functions", () => {
+      // eslint-disable-next-line max-nested-callbacks
```

Comment:
I had a choice here between:
* Having two tests that exceed max level of nesting (what I did)
* Having two tests with really long names that exceed the max line length.
* Having two tests with short but rather cryptic names.

I felt I picked the lesser evil but I'm open to being told otherwise. :)

Reply:
haha - i dont care you can always bump the max level of nesting